src/main/groovy/de/dkfz/roddy/bam2fastq/BamToFastqWorkflow.groovy

```diff
@@ -172,20 +162,23 @@ class BamToFastqWorkflow extends Workflow {
         for (BaseFile bamFile : bamFiles) {
 
             if (config.outputPerReadGroup) {
-                List<String> readGroups = readGroupsPerBamfile[bamFile.absolutePath]
+                ReadGroupGroup groups = readGroupsPerBamfile[bamFile.absolutePath]
 
-                FileGroup unsortedFastqs = bam2fastq(bamFile, readGroups)
+                ReadGroupGroup groupsWithUnsortedFastqs = bam2fastq(bamFile, groups)
 
                 if (config.sortFastqs) {
-                    for (int i = 0; i < readGroups.size(); i++) {
-                        this.sortFastqs(bamFile.getPath().name, readGroups[i], unsortedFastqs[i * 2], unsortedFastqs[i * 2 + 1]) // All paired end, sorted by read group.
+                    for (ReadGroup grp : groupsWithUnsortedFastqs.readGroups) {
+                        sortPairedFastqs(bamFile.getPath().name, grp.name, grp[ReadFileType.READ1], grp[ReadFileType.READ2]) // R1/R2
+                        sortFastq(bamFile.getPath().name, grp.name, grp[ReadFileType.UNMATCHED_READ1]) // U1
+                        sortFastq(bamFile.getPath().name, grp.name, grp[ReadFileType.UNMATCHED_READ2]) // U2
+                        sortFastq(bamFile.getPath().name, grp.name, grp[ReadFileType.SINGLETON]) // S
                     }
                 }
 
             } else {
 //                FileGroup out = bam2fastq(cfg, BaseFile)
```

Comment:
Remove comments

Reply:
Maybe just state, that this alternative is not implemented yet? You could check that earlier and skip execution at all.